### PR TITLE
RHCLOUD-7520 Extract System Profile schema to a separate file

### PR DESCRIPTION
**Pull request by Glutexo**
Extracted System Profile schema from the OpenAPI specification file. [JSON Reference](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03) is used to link the external schema to the main document.

[prance](https://pypi.org/project/prance/) library used to dereference and compile the schema into a single specification object.

For the schema to be valid for OpenAPI, the _/$def/SystemProfile/$id_ property had to be removed. I’ll remove it from _inventory-schemas_ too. Also the _x-body-name_ Connexion proprietary parameter had to be removed.

A part of [RHCLOUD-7520](https://projects.engineering.redhat.com/browse/RHCLOUD-7520).

### CODE REVIEW

**Comment by Glutexo** [comment]
Hmmm… Facts tests are broken. Let’s see why.

**Comment by Glutexo** [comment]
Okay. The culprit is that _prance_ doesn’t recognize a proprietary _x-body-value_ Connexion attribute. Sweet.

**Comment by Glutexo** [comment]
Fixed.

File: app/__init__.py [hunk 1]
```diff
@@ -35,9 +40,16 @@ def inventory_config():
 
 
 def create_app(runtime_environment):
+    # This feels like a hack but it is needed.  The logging configuration
+    # needs to be setup before the flask app is initialized.
+    configure_logging()
+
+    app_config = Config(runtime_environment)
```
Comment (jharting) [inline app/__init__.py:47]: is this supposed to be done twice?

File: app/__init__.py [hunk 2]
```diff
@@ -35,9 +40,16 @@ def inventory_config():
...
+    app_config.log_configuration()
+
     connexion_options = {"swagger_ui": True}
 
-    # This feels like a hack but it is needed.  The logging configuration
+    # This feels like a hack but it is needed.  The loggingconfiguration
```
Comment (jharting) [inline app/__init__.py:52]: loggingconfiguration?

Comment (Glutexo) [reply app/__init__.py:52]: Hehe.

File: app/__init__.py (hunk 1 above)
Comment (Glutexo) [reply app/__init__.py:47]: Oh noez, rebase conflict artifact.

Comment (jharting) [reply app/__init__.py:47]: @Glutexo This conversation is marked resolved but nothing changed. Did you forget to git push?

Comment (Glutexo) [reply app/__init__.py:47]: Resolved as in “All understood” not as in “Fixed and pushed”. 😃

**Comment by Glutexo** [comment]
Thanks for spotting out those dumb bugs, @jharting. Hopefully fixed now!

File: logconfig.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ loggers:
     - logstash
     propagate: true
   sqlalchemy.engine:
-    level: WARNING
+    level: INFO
```
Comment (jharting) [inline logconfig.yaml:25]: :warning: :arrow_right: :information_source: :question:

Comment (Glutexo) [reply logconfig.yaml:25]: Good catch, thanks!

**Comment by Glutexo** [comment]
Fixed that log configuration change committed by mistake. I wonder whether there is a way to ignore changes to files already committed to repository. Something like _.gitignore_, but for existing files.

**Comment by Glutexo** [comment]
🎉  Yay. Thanks!

**Pull request merged**